lib/tools/apk-utils.js

```diff
@@ -509,12 +540,12 @@ apkUtilsMethods.setDeviceLanguageCountry = async function (language, country) {
   if (apiLevel < 23) {
     let curLanguage = await this.getDeviceLanguage();
     let curCountry = await this.getDeviceCountry();
-    if (hasLanguage && language !== curLanguage) {
-      await this.setDeviceLanguage(language);
+    if (hasLanguage && language.toLowerCase() !== curLanguage.toLowerCase()) {
```

Comment:
With all these `toLowerCase` calls, if it doesn't matter in the setting (as it doesn't seem to from the implementation here), why not just do it once for everything and have the variables hold the lower case versions? Since strings are passed by value it won't affect the calling code's version of `language` and `country`.
```js
language = language.toLowerCase();
let curLanguage = (await this.getDeviceLanguage()).toLowerCase();
// ...
```

Reply:
`curLanguage` and `curLanguage` can `.toLowerCase()`, but:

- `language` and `country` can be undefined in `apiLevel < 23` and `apiLevel === 23` case.
- `language.toLowerCase()` and `country.toUpperCase()` is called once in `else`.
- `locale.toLowerCase() !== curLocale.toLowerCase()` is called only comparition to make the comparition in case insentive following Android's `Locale`.
    - Is there any method to compare two string in case insentive in JavaScript? I haven't been able to find such method yet... 

Thus, I call `.toLowerCase()` or `.toUpperCase()` when they are needed.

Reply:
Since the empty case is handled by `hasLanguage`, then `language = (language || '').toLowerCase()` would do it.